src/hazelcore/dynamics/DHExp.re

```diff
@@ -207,7 +207,7 @@ let rec mk_tuple: list(t) => t =
   | [d, ...ds] => Pair(d, mk_tuple(ds));
 
 let cast = (d: t, t1: HTyp.t, t2: HTyp.t): t =>
-  if (HTyp.eq(t1, t2)) {
+  if (HTyp.eq(t1, t2) || t2 == ModeSwitch) {
```

Comment:
if we're treating ModeSwitch as an unknown for consistency, then we shouldn't need this anymore I think